### ISSUE TITLE
Request system/clipboard bug

### ISSUE THREAD

**Issue by Detruo** [ldtteam/minecolonies#10497]
### Is there an existing issue for this?

- [x] I have searched the existing issues.

### Are you using the latest MineColonies Version?

- [X] I am running the latest **beta/release** version of MineColonies for my Minecraft version.
I am also running the latest versions of other mods that are part of my problem.

### Did you check on the Wiki? or ask on Discord?

- [X] I checked the MineColonies Wiki and made sure my issue is not covered there. Or I was sent from discord to open an issue here.

### What were you playing at the time? Were you able to reproduce it in both settings?

- [ ] Single Player
- [X] Multi Player

### Minecraft Version

1.20

### MineColonies Version

1.20.1-1.1.760

### Structurize Version

1.20.1-1.0.760

### Related Mods and their Versions

1. Forge version: 47.3.11

### Current Behavior

After some time clipboard stopped showing NPC requests. The command "/mc colony requestsystem-reset-all" sometimes works but for a few seconds. Sometimes when tries to add an NPC custom recipe via request or I open the clipboard and it crashes the game. When I want to request something via Postbox, the request doesn't appear on the right side, but the NPC still brings the requested items.

### Expected Behavior

X

### Reproduction Steps

X

### Logs

https://gist.github.com/Detruo/df5d47d16314d8d13c79a48676d86785

### Footer

---
## Viewers
* Add a thumbs-up to the bug report if you are also affected. This helps the bug report become more visible to the team and doesn't clutter the comments.
* Add a comment if you have any insights or background information that isn't already part of the conversation.

**Comment by kezmodius** [comment]
A client on our server is having a similar crash.
 Minecolonies 1.20.1-1.1.762-snapshot
 Structurize 1.20.1-1.0.760-snapshot
 BlockUI 1.20.1-1.0.186-beta
Crash report: https://gist.github.com/kezmodius/c17e5d037aebb41ee25c86e9f89e7561
Latest client log: https://gist.github.com/kezmodius/55b06cabb7b79c593f2a2435139088ab

I can get you server logs if you want them.

**Comment by kezmodius** [comment]
We did a request system reset. Things worked after that until log out.
Came back the next day and still crashing.
https://gist.github.com/kezmodius/598f65df09775302f93d9ab234c39aa8

**Comment by Raycoms** [comment]
Is there any serverside log that appears shortly before that?

**Comment by kezmodius** [comment]
I'll ask the owner for it. 
The player says the video here is the issue they're having: 
https://github.com/ldtteam/minecolonies/issues/10447

**Comment by Detruo** [comment]
This is an error while server is starting (part of server logs)
https://gist.github.com/Detruo/bf36fe03a01a7d1903d8e208de32fbaf

**Comment by Raycoms** [comment]
okay, that is also a symptom of the same thing. There must be an error somewhat earlier causing it.

**Comment by kezmodius** [comment]
Server logs for the 4th and 5th of December. Player reported issues on the 5th.
 
https://gist.github.com/kezmodius/b30d44564b96748938da0273d5e90f53
https://gist.github.com/kezmodius/2475c586d0868c406cdc8f509f80583e

**Comment by Raycoms** [comment]
Odd, logs have nothing out of the ordinary. Were mods removed or so?

**Comment by Detruo** [comment]
In my case, nothing was deleted. I'm still playing with the "Minecolonies: Dimensional Adventure 2.1.6" modpack. Updates don't bring any results. The request system started to break down after a few days of playing (maybe even a week or two).
Recently, I've noticed that from time to time the system works for a few seconds. Honestly, opening the clipboard is a roulette: it either works or it doesn't, sometimes it can crash the game.

**Comment by Raycoms** [comment]
Yeah, something corrutps the request system on the serverside which causes issues on the client then. If you can do a request system reset. and then upload the serverside log the moment things break again, that would be great.

**Comment by Detruo** [comment]
I just changed the "colonyloadstrictness" setting to 1 in minecolonies-server.toml and it seems to have fixed the problem. The request system has been working fine for a while now. My colony is very large and it is possible that some chunks were not loading. It is possible that this is a coincidence, but I think it is worth mentioning.

**Comment by Raycoms** [comment]
An unloaded building can cause issues indeed. But a log would still be great to be able to check this out =D

**Comment by Detruo** [comment]
I restarted the request system at 19:38 and it worked the whole time. Now it started breaking again since the message "Handling Survival Placement in Colony" appeared

https://gist.github.com/Detruo/5275f5ba841d6397c80a5463fbe3e7bb

I get an error on the client every second:
https://gist.github.com/Detruo/6d3c01dfebafc449382a0adaf78501c5

**Comment by AnyankaT** [comment]
> A client on our server is having a similar crash. Minecolonies 1.20.1-1.1.762-snapshot Structurize 1.20.1-1.0.760-snapshot BlockUI 1.20.1-1.0.186-beta Crash report: https://gist.github.com/kezmodius/c17e5d037aebb41ee25c86e9f89e7561 Latest client log: https://gist.github.com/kezmodius/55b06cabb7b79c593f2a2435139088ab
> 
> I can get you server logs if you want them.

Here is more information for the ongoing issue with the clipboard. It's only one colony, but it's anyone in the colony - both the player who owns it, and me (server admin) added as officer. 

At this point his colony is unplayable, but I'm  unsure whether the best course of action is to update the server (which has nearly 60 semi-active players who will need to update their modpacks) or have him restart his colony, since it's ONLY his colony where this is happening. 

Attached are the server latest.log, my client latest.log, and the colony export, just after the issue happened for me and the request system was reset.

[latest-server.log](https://github.com/user-attachments/files/18095784/latest-server.log)

[latest-client.log](https://github.com/user-attachments/files/18095786/latest-client.log)
[colony16Export.zip](https://github.com/user-attachments/files/18095789/colony16Export.zip)

**Comment by AnyankaT** [comment]
I updated our test server to 771 and it's still happening.
[latest.log](https://github.com/user-attachments/files/18097293/latest.log)

**Comment by AnyankaT** [comment]
[latest.log](https://github.com/user-attachments/files/18097731/latest.log)

There's an error even when not in the colony.

**Comment by Raycoms** [comment]
Yes, this error is quite clear, it's actually loaded from file. I wonder if there is an error before this occurs.

**Comment by AnyankaT** [comment]
How could I find that?

**Comment by Raycoms** [comment]
Do a request system reset, wait until you see the error in the log. Send me the error that spans over this time (I couldn't find the reset in the previois log)

**Comment by AnyankaT** [comment]
Started both server and client, reset the request system, clipboard worked (showed ancient tome and smeltable ore) then broke. Shut down server and pulled both logs.
[latest.log](https://github.com/user-attachments/files/18099634/latest.log)
[latest-server.log](https://github.com/user-attachments/files/18099638/latest-server.log)

**Comment by Raycoms** [comment]
Can you check if there is a crafter that has a recipe that either has air as input, or output?

**Comment by skadlig** [comment]
Could it also be a request for a minimum stock of air? I might have seen one of those in my colony. 
I will check as soon as the kids are in bed and the washing up is done.

**Comment by Raycoms** [comment]
Yes, minimum stock is similarily possible

**Comment by skadlig** [comment]
I couldn't find any of those instances (unfortunately?). So I guess my issue is caused by something else, less I missed a crafter/builder. 

The closest oddity I got is a courier who seems to be hired at two workshops at once and at the same time non of them. Ie the workers tab in both warehouses lists his name but when pressing "Manage workers" his name is missing. Considering couriers part in the request system I guss it could be connected?

**Comment by ATXero** [comment]
Hii! 
I'm the user on @AnyankaT and @kezmodius  server having clipboard bugs and sudden crashes, i was told to log in the game, reset the request system, wait for the bug to appear, close and send the latest log. So here i am.
[latest.log](https://github.com/user-attachments/files/18102597/latest.log)

**Comment by Raycoms** [comment]
Can you update to latest and keep me updated?

**Comment by ATXero** [comment]
@AnyankaT said that even in the latest update (771) on the test server was still having this bug

**Comment by MotionlessTrain** [comment]
It has not been solve yet in 1.1.772, but the logging has been improved in that version, which can help Ray find out what the actual issue is

**Comment by Raycoms** [comment]
Yes, please send me the newest server & client logs

**Comment by Detruo** [comment]
I updated the mod to version 773. I started the server and restarted request system. After a while I started getting errors in the client logs, but the requests were still visible on the clipboard. I also noticed after a while that Cowhand was requesting 128 of Air - exactly the same amount as I set him to milk cows during the day. The minimum Stock is not set for him.
Client logs: https://gist.github.com/Detruo/37375eae31b0fc7509485c368c433d9b
Server logs: https://gist.github.com/Detruo/c7a06d376968b3bf11e3b5a4873d3fcd

![image](https://github.com/user-attachments/assets/f23f806c-f4c2-4205-b83d-30fee53ef3cd)

**Comment by AnyankaT** [comment]
Same thing happening here. I updated to 774. Cowhand was requesting 1000 air, same as the milk attempts that were set. 

It's interesting to note that the clipboard did not bug out this time. I cancelled the request for air, but it did return.

[latest.log](https://github.com/user-attachments/files/18117482/latest.log)
[latest-server.log](https://github.com/user-attachments/files/18117490/latest-server.log)

![image1](https://github.com/user-attachments/assets/5bb9c5fb-0122-4b2b-a153-60b4d03bc028)
![image2](https://github.com/user-attachments/assets/2284ba56-6339-45d2-88ad-bfda312fa670)

**Comment by Raycoms** [comment]
This is very weird. It recognizes it as a bottle request, because it's resolved correctly, but it's still air?

**Comment by Raycoms** [comment]
Nvm, the problem is on the client, on the server the data is correctly actually

**Comment by Raycoms** [comment]
Do you have any datapacks on the client that might be messing with the bottles?

**Comment by AnyankaT** [comment]
Not that I'm aware of, but I will go back through and check our data packs.

Michelle Smith
Independent Language Trainer

**Comment by AnyankaT** [comment]
No, there are no datapacks in our modpack that touch the bottle recipes. The player is using our pack as-is with no additions.

**Comment by Raycoms** [comment]
Is this still an issue?

**Comment by AnyankaT** [comment]
I think this one has been fixed.

Michelle Smith
Independent Language Trainer
